src/main/scala/io/iohk/ethereum/vm/ProgramContext.scala

```diff
@@ -4,23 +4,23 @@ import io.iohk.ethereum.domain._
 
 
 object ProgramContext {
-  def apply(stx: SignedTransaction, blockHeader: BlockHeader, world: WorldStateProxy): ProgramContext = {
+  def apply[W <: WorldStateProxy[W, S], S <: Storage[S]](stx: SignedTransaction, blockHeader: BlockHeader, world: W): ProgramContext[W, S] = {
     import stx.tx
 
     val senderAddress = stx.recoveredSenderAddress.get // FIXME: get, it should be validated but...
-    val (world1, recipientAddress, program) = callOrCreate(world, tx, senderAddress)
+    val (world1, recipientAddress, program) = callOrCreate[W, S](world, tx, senderAddress)
 
     val env = ExecEnv(recipientAddress, senderAddress, senderAddress, tx.gasPrice, tx.payload,
       tx.value, program, blockHeader, callDepth = 0)
 
-    ProgramContext(env, tx.gasLimit, world1)
+    ProgramContext(env, tx.gasLimit, world1.asInstanceOf[W])
```

Comment:
:+1: I was quick with this, and trying several things... There may be more little omissions like this